HIP/hip-358.md

```diff
@@ -0,0 +1,123 @@
+---
+hip: 358
+title: Allow Token Create Through Hedera Token Service Precompiled Contract
+author: Stoyan Panayotov <stoyan.panayotov@limechain.tech>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Last Call
+last-call-date-time: 2022-03-30T07:00:00Z
+created: 2022-02-09
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/375
+---
+
+## Abstract
+
+Describe the integration of Hedera Token Service (HTS) with the Hedera Smart Contract Service (HSCS), allowing contracts to create HTS tokens programmatically.
+
+## Motivation
+
+Allow smart contract developers on Hedera to create HTS fungible and non-fungible tokens through the HTS precompiled contract giving smart contracts developers on Hedera to ability to leverage Hedera native token functionality.
+
+## Rationale
+
+HIP-206 already allows token `mint`, `burn`, `associate`, `dissociate` and `transfer` through the HTS precompiled contract. Allowing creation of tokens via smart contracts is a natural extension of the precompiled contract and would enable smart contract developers to rely solely on contract code for token management. 
+
+## User stories
+
+As a smart contract developer, I want to be able to create a native HTS fungible token through a solidity contract call.
+
+As a smart contract developer, I want to be able to create a native HTS fungible token with custom fees through a solidity contract call.
+
+As a smart contract developer, I want to be able to create a native HTS non-fungible token through a solidity contract call.
+
+As a smart contract developer, I want to be able to create a native HTS non-fungible token with custom fees through a solidity contract call.
+
+## Specification
+
+### EVM Precompile extension
+
+The [Solidity file for development](../assets/hip-206/solidity/IHederaTokenService.sol) is updated with new types and function signatures that the precompile will respond to. It is included in this HIP by reference.
+
+### Solidity Function Signatures 
+
+The following structs are defined to simplify the interface between solidity and the HTS precompile: 
```

Comment:
For the data structures we should describe what each parameter does, not just the type.  Probably in a markdown prose section not in the table.